### PR TITLE
[Bug] Fix bug that wrong exception message returned by insert statement

### LINKED ISSUE #7831: [Bug] Wrong exception message returned by insert statement
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master branch (commit id : 0efef1b332300887ee0473f9df9bdd9d7297d824)

### What's Wrong?

`insert` statement may return exception message `Execute timeout` after load data failed. But the real reason is that there exists unhealthy backend, not execute timeout.
```
ERROR 1105 (HY000): errCode = 2, detailMessage = Execute timeout
```

### What You Expected?

Right exception message is returned after `insert` failed. 

insert failed because of unhealthy backend:
```
MySQL [ssb]> insert into lineorder_flat select * from lineorder_flat;
ERROR 1105 (HY000): errCode = 2, detailMessage = There exists unhealthy backend. backend 10002 is down
```

insert failed because of timeout:
```
MySQL [ssb]> set query_timeout=30;
Query OK, 0 rows affected (0.01 sec)

MySQL [ssb]> insert into lineorder_flat select * from lineorder_flat;
ERROR 1105 (HY000): errCode = 2, detailMessage = Execute timeout
```

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by weizuo93**
# Proposed changes

Issue Number: close #7831

## Problem Summary:

`insert` statement may return exception message `Execute timeout` after load data failed. But the real reason is that there exists unhealthy backend, not execute timeout.
```
MySQL [ssb]> insert into lineorder_flat select * from lineorder_flat;
ERROR 1105 (HY000): errCode = 2, detailMessage = Execute timeout
```

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

### CODE REVIEW

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -1248,10 +1248,15 @@ private void handleInsertStmt() throws Exception {
 
                 coord.exec();
 
-                coord.join(context.getSessionVariable().getQueryTimeoutS());
+                boolean notTimeout = coord.join(context.getSessionVariable().getQueryTimeoutS());
                 if (!coord.isDone()) {
                     coord.cancel();
-                    ErrorReport.reportDdlException(ErrorCode.ERR_EXECUTE_TIMEOUT);
+                    if (notTimeout) {
+                        ErrorReport.reportDdlException(ErrorCode.ERR_UNHEALTHY_BACKEND_EXISTS);
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1255]: Using query state is better

Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:1255]: Or can you print out the wrong backend id through the error code

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 1 above)
Comment (weizuo93) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: > Or can you print out the wrong backend id through the error code

OK, thanks for your review.

**Review by morningman — APPROVED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**